lib/action_cable/channel/base.rb

```diff
@@ -75,6 +75,41 @@ class Base
       attr_reader :params, :connection
       delegate :logger, to: :connection
 
+      class << self
+        # A list of method names that should be considered actions. This
+        # includes all public instance methods on a channel, less
+        # any internal methods (defined on Base), adding back in
+        # any methods that are internal, but still exist on the class
+        # itself.
+        #
+        # ==== Returns
+        # * <tt>Set</tt> - A set of all methods that should be considered actions.
+        def action_methods
+          @action_methods ||= begin
+            # All public instance methods of this class, including ancestors
+            methods = (public_instance_methods(true) -
+              # Except for public instance methods of Base and its ancestors
+              ActionCable::Channel::Base.public_instance_methods(true) +
+              # Be sure to include shadowed public instance methods of this class
+              public_instance_methods(false)).uniq.map(&:to_s)
+            methods.to_set
+          end
+        end
+
+        # action_methods are cached and there is sometimes need to refresh
```

Comment:
Sometimes means that someone might access `action_methods` (and caching action methods)  before defining new methods. But I don't see any cases where one would call this manually (it's called automatically by `method_added`). I'm going to make this methods protected.